data/api/server-server/joins-v2.yaml

```diff
@@ -207,9 +207,9 @@ paths:
                     title: SignedMembershipEvent
                     x-addedInMatrixVersion: "1.2"
                     description: |-
-                      Required if the room version [supports restricted join rules](/rooms/#feature-matrix). The signed
-                      copy of the membership event sent to other servers by the resident server, including the resident
-                      server's signature.
+                      Required if the `content` of the event in the request contained the `join_authorised_via_users_server`
+                        field. The signed copy of the membership event sent to other servers by the resident server,
```

Comment:
As I've just written over at #1708: this doesn't seem to match Synapse's behaviour. AFAICT, the presence of `join_authorised_via_users_server` is not sufficient for Synapse to return a (signed) event here.

Reply:
The synapse code is at https://github.com/element-hq/synapse/blob/v1.108.0/synapse/federation/federation_server.py#L952-L956. 

Currently, it returns the event for all requests to `/send_join`. But it only *signs* the event if the user is actually performing a restricted join.